/*******************************************************************************
 * Copyright (c) 2014 Elektrobit Automotive GmbH
 * All rights reserved
 *
 * This file is part of project EB RACE
 *
 * Unauthorized copying of this file is strictly prohibited
 *******************************************************************************/
package TargetAgent.Protocol.NetworkPacketSniffer;
option java_package = "com.elektrobit.ebrace.targetdata.adapter.networkpacketsniffer";
option java_outer_classname = "NetworkPacketSnifferTAProto";

enum MESSAGE_SOURCE 
{
	MESSAGE_SOURCE_WEBSOCK	 			 = 0;
	MESSAGE_SOURCE_SOMEIP				 = 1;
	MESSAGE_SOURCE_CORBA				 = 2;
}
message TCPHeader{
	required uint64	src_port = 1;
	required uint64	dst_port = 2;
}
message UDPHeader{
	required uint64	src_port = 1;
	required uint64	dst_port = 2;
}

message NetworkPacket {
	required MESSAGE_SOURCE source = 1;
	required string pdu_payload = 2;
	required string src_addr = 3;
	required string dst_addr = 4;
	optional TCPHeader tcp = 5;
	optional UDPHeader udp = 6;
}

